components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__genes.xml

```diff
@@ -120,105 +120,79 @@
-             var info = data &amp;&amp; data.response &amp;&amp; data.response[geneName] &amp;&amp; data.response[geneName].external_ids;
-             if (info) {
+  }
 
-                var geneLinks = element.next('div.gene-links') || new Element('div', {'class' : 'gene-links'});
-                var externalDBs = [
-                          { name : 'GENECARDS', url : 'http://www.genecards.org/cgi-bin/carddisp.pl?gene=',          field : 'genecards_id'},
-                          { name : 'OMIM',      url : 'http://www.omim.org/entry/',                                  field : 'omim_id'},
-                          { name : 'Entrez',    url : 'http://www.ncbi.nlm.nih.gov/gene/?term=',                     field : 'entrez_id'},
-                          { name : 'RefSeq',    url : 'http://www.ncbi.nlm.nih.gov/nuccore/',                        field : 'refseq_accession'},
-                          { name : 'Ensembl',   url : 'http://useast.ensembl.org/Homo_sapiens/Gene/Compara_Tree?g=', field : 'ensembl_gene_id'}
-                ];
-                externalDBs.each(function (item) {
-                         var value = info[item.field];
-                         if (value) {
-                           if (!value.each) {
-                             value = [value];
-                           }
-                           value.each (function (id) {
-                             geneLinks.insert(new Element('a', {
-                               'href'   : item.url + id,
-                               'class'  : 'button secondary'
-                             }).update(item.name + ': ' + id).wrap('span', {'class' : 'buttonwrapper'}));
-                           });
-                         }
-                });
-                geneLinks.select('a').each(function(item) {
-                         item.observe('click', function(event) {
-                            event.stop();
-                            window.open(item.href);
-                         })
-                });
-                if (!element.__handleEmptyValue) {
-                   element.__handleEmptyValue = true;
-                   ['keyup', 'change', 'keypress'].each(function(eventName) {
-                     element.observe(eventName, function (event) {
-                       if (element.value == '') {
-                         geneLinks &amp;&amp; geneLinks.update('');
-                       }
-                     })
-                   });
-                }
-             }
-           });
-         },
-         onComplete : function () {
-           geneRefs.each(function(element) {
-             var geneLinks = element.next('div.gene-links');
-             if (geneLinks) {
-               geneLinks.removeClassName('loading');
-             }
-           });
-         }
-       }
-    );
+  var updateInfo = function(el, value, isSuggestSelected) {
+    var newValue = (!value || value == "") ? el.value : value;
```

Comment:
`||` is a binary operator, just like `+`, `-`, or `*`. And just like those, it takes two operands and returns one value, and they can be used anywhere where an expression is evaluated. And in JavaScript, an expression doesn't have to be assigned to be valid, you can just write `2+3;` as a statement, it will evaluate to `5` and it doesn't care that that 5 is lost/discarded.

In JavaScript, all operators accept any type of value, and works with those values somehow behaving like the expected type of value. For example, `"5" - "3"` returns `2`, even if the inputs are strings.

So boolean operators evaluate each value as either `true` or `false`, without actually converting those values to an actual boolean; thus values in JavaScript are called "truthy" or "falsey".

Unlike the boolean logic tables that are usually taught, where `false || true = true`, JavaScript evaluates boolean expressions in terms of its inputs, using short circuiting: the output of a boolean operator is one of the two inputs:

- falsey && anything = falsey (left operand)
- truthy && something = something (right operand)
- truthy || anything = truthy (left operand)
- falsey || something = something (right operand)